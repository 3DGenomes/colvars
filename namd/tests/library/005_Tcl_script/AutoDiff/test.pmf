# 1
#       11.9       0.2       101  0

 12  763.152
 12.2  777.148
 12.4  671.873
 12.6  568.532
 12.8  467.336
 13  369.712
 13.2  278.713
 13.4  196.25
 13.6  124.307
 13.8  64.7711
 14  22.3458
 14.2  2.77288
 14.4  0
 14.6  0
 14.8  0
 15  0
 15.2  0
 15.4  0
 15.6  0
 15.8  0
 16  0
 16.2  0
 16.4  0
 16.6  0
 16.8  0
 17  0
 17.2  0
 17.4  0
 17.6  0
 17.8  0
 18  0
 18.2  0
 18.4  0
 18.6  0
 18.8  0
 19  0
 19.2  0
 19.4  0
 19.6  0
 19.8  0
 20  0
 20.2  0
 20.4  0
 20.6  0
 20.8  0
 21  0
 21.2  0
 21.4  0
 21.6  0
 21.8  0
 22  0
 22.2  0
 22.4  0
 22.6  0
 22.8  0
 23  0
 23.2  0
 23.4  0
 23.6  0
 23.8  0
 24  0
 24.2  0
 24.4  0
 24.6  0
 24.8  0
 25  0
 25.2  0
 25.4  0
 25.6  0
 25.8  0
 26  0
 26.2  0
 26.4  0
 26.6  0
 26.8  0
 27  0
 27.2  0
 27.4  0
 27.6  0
 27.8  0
 28  0
 28.2  0
 28.4  0
 28.6  0
 28.8  0
 29  0
 29.2  0
 29.4  0
 29.6  0
 29.8  0
 30  0
 30.2  0
 30.4  0
 30.6  0
 30.8  0
 31  0
 31.2  0
 31.4  0
 31.6  0
 31.8  0
 32  0

