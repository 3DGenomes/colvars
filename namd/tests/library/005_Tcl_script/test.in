colvarsTrajFrequency    2


colvar {
    name d

    lowerBoundary 12
    upperBoundary 32
    width 0.2

    distance {
        group1 { atomnumbers 4 }
        group2 { atomnumbers 99 }
    }
}


abf {
    colvars        d
    fullSamples    0
    outputFreq     60
}


harmonic {
    colvars d
    centers 12
    forceConstant 10

    outputEnergy
}
