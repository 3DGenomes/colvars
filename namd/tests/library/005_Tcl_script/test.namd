# -*- tcl -*-

# FILENAMES
set outName             [file rootname [file tail [info script]]]
#set inFileNum           [expr [scan [string range $outName end-1 end] "%d"] - 1]
#set inName              [format "%s%02u" [string range $outName 0 end-2] $inFileNum]

# FILES

#outputname              test01
#DCDfile                 test01.dcd

# MD SETUP
timestep          0.5
numSteps		  20

# Interactive molecular dynamics
#IMDon               on
#IMDport             3000
#IMDfreq             20
#IMDwait             on

colvars			  on
colvarsConfig	  test.in
#colvarsInput  $inName.colvars.state
#colvarsInput  test.colvars.state

#tclForces on
#tclForcesScript {
#  addatom 12
#  proc calcforces {} {
#    addforce 12 { 10 0 0 }
#  }
#}

# INPUT
structure               ../Common/da.psf 
#parameters              missing.par
parameters              ../Common/par_all22_prot.inp
paraTypeCharmm          on
coordinates             ../Common/da.min.pdb

temperature 250
#bincoordinates       $inName.coor
#binvelocities        $inName.vel

# OUTPUT

outputname              ${outName}
outputenergies          100
outputtiming            2000
binaryoutput            no
# binaryrestart           yes

#XSTFreq                 1000

# DCD TRAJECTORY

DCDfile                 ${outName}.dcd
DCDfreq                 2000

seed		87654321

# CONSTANT-T
langevin                on
langevinTemp		    300.0
langevinDamping         10.0
langevinHydrogen	    on

# SPACE PARTITIONING
splitpatch              hydrogen
hgroupcutoff            2.8
stepspercycle           20
margin                  1.0

# CUT-OFFS
switching               on
switchdist              9.0
cutoff                  11.0
pairlistdist            12.0

# RESPA 
fullElectFrequency      1
nonbondedFreq           1

# 1-4 NON-BONDED
exclude                 scaled1-4
1-4scaling              1.0

# COM
commotion               no

# SHAKE
rigidBonds              none

proc runcmd { cmd } {
  print "Running $cmd"
  print [eval $cmd]
}

run 0

foreach c [colvars list] { print "$c = [colvars colvar $c value]"}        
foreach b [colvars list biases] { print "E($b) = [colvars bias $b energy]"}

runcmd "colvars update"

foreach c [colvars list] { print "$c = [colvars colvar $c value]"}        
foreach b [colvars list biases] { print "E($b) = [colvars bias $b energy]"}

run 20

foreach c [colvars list] { print "$c = [colvars colvar $c value]"}        
foreach b [colvars list biases] { print "E($b) = [colvars bias $b energy]"}

# colvars config "colvar {
#   name d2
#   distance {
#     group1 { atomnumbers 4 }
#     group2 { atomnumbers 99 }
#   }
# }"
# runcmd "colvars colvar d2 value"
# runcmd "colvars update"
# runcmd "colvars colvar d2 value"
# run 20
# runcmd "colvars colvar d2 value"
# runcmd "colvars colvar d2 delete"

colvars config "harmonic {
  name harm2
  colvars d
  centers 13
  forceConstant 1.
  outputEnergy yes
}"
run 20 

runcmd "colvars bias harmonic1 delete"

runcmd "colvars bias harm2 energy"
runcmd "colvars bias harm2 update"
run 20 
runcmd "colvars bias harm2 energy"
runcmd "colvars bias harm2 delete"

colvars reset
# reinitialize just for the fun of it ;-)
colvars configfile 	  test.in
