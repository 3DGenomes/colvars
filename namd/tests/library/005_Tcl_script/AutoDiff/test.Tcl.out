TCL: d = 0.00000000000000e+00
TCL: E(abf1) = 0
TCL: E(harmonic1) = 0
TCL: Running cv update
TCL: 
TCL: d = 0.00000000000000e+00
TCL: E(abf1) = 0
TCL: E(harmonic1) = 18000
TCL: Original numsteps 20 will be ignored.
TCL: Running for 20 steps
TCL: d = 1.28684277602171e+01
TCL: E(abf1) = 0
TCL: E(harmonic1) = 94.2708
TCL: Running for 20 steps
TCL: Running cv bias harmonic1 delete
TCL: 
TCL: Running cv bias harm2 energy
TCL: 17.8166
TCL: Running cv bias harm2 update
TCL: 17.8166
TCL: Running for 20 steps
TCL: Running cv bias harm2 energy
TCL: 13.4817
TCL: Running cv bias harm2 delete
TCL: 
